primitives/messages/src/lib.rs

```diff
@@ -237,12 +237,12 @@ impl<Result> ReceivedMessages<Result> {
 
 /// Result of single message receival.
 #[derive(RuntimeDebug, Encode, Decode, PartialEq, Eq, Clone, TypeInfo)]
-pub enum ReceivalResult {
+pub enum ReceivalResult<DispatchLevelResult: Clone + Decode + sp_std::fmt::Debug + Eq> {
```

Comment:
Can this be avoided by replacing with `#[derive(RuntimeDebugNoBound, Encode, Decode, PartialEqNoBound, EqNoBound, CloneNoBound, TypeInfo]`?

Reply:
hmm, I just removed those `Clone + Decode + sp_std::fmt::Debug + Eq` from ReceivalResult and MessageDispatchResult and they are not needed here.
I have no idea why I added them there :)